src/test/ui/generic-associated-types/bugs/hrtb-implied-1.stderr

```diff
@@ -0,0 +1,20 @@
+error[E0716]: temporary value dropped while borrowed
+  --> $DIR/hrtb-implied-1.rs:33:22
+   |
+LL |     let slice = &mut ();
+   |                      ^^ creates a temporary which is freed while still in use
+...
+LL |     print_items::<WindowsMut<'_>>(windows);
+   |     -------------------------------------- argument requires that borrow lasts for `'static`
+LL | }
+   | - temporary value is freed at the end of this statement
+   |
+note: due to current limitations in the borrow checker, this implies a `'static` lifetime
+  --> $DIR/hrtb-implied-1.rs:28:26
+   |
+LL |     for<'a> I::Item<'a>: Debug,
```

Comment:
@estebank maybe you have some thoughts on how to make this point to the whole clause?

Reply:
I thought a bit about this since you had mentioned it earlier -- it's probably not generalizable to point to the whole clause because we can have things like `where Ty: Trait + OtherTrait` -- the choice for the spans in `predicates_of` to point to just the trait part of the predicate was probably made for that reason.

Reply:
Right, it's not simply enough to say "point to the entire clause. Instead, I'm imagining something like:

```
for<'a> T: Trait<'a>
^^^^^^^^^^^^^^^^^^^^

for<'a> T: Send + Trait<'a>
^^^^^^^^^^        ^^^^^^^^^

for<'a> T: Trait<'a> + Send
^^^^^^^^^^^^^^^^^^^^
```

Reply:
The problem is that we currently have no way (that I know of) to even get that span info from an `InstantiatedPredicate`. We lose that when generating the `GenericPredicates`, and there's no mapping back, AFAICT. I was looking a bit at instead storing like the owner and which predicate it is, but that also doesn't quite work, since lowering is done in a bit more complicated fashion.